runtime/near-vm-logic/src/logic.rs

```diff
@@ -2162,7 +2162,16 @@ impl<'a> VMLogic<'a> {
         let buf = self.get_vec_from_memory_or_register(ptr, len)?;
         self.gas_counter.pay_base(utf8_decoding_base)?;
         self.gas_counter.pay_per(utf8_decoding_byte, buf.len() as u64)?;
-        let account_id = AccountId::from_utf8(buf).map_err(|_| HostError::BadUTF8)?;
+
+        // We return an unsafely constructed AccountId here for the sole purpose of
+        // ensuring backwards compatibility, we go ahead to validate this further down the line
+        // in node-runtime/verifier.rs#fn(validate_receipt)
```

Comment:
This is a bit of a misleading comment, as the `validator_stake` code path uses this function as well, and doesn't do any validation.

Reply:
Agree

Reply:
Yeah, but the previous behavior in `validator_stake` didn't require validation either so it's practically fine. Just updated the comment and we're good.